projects/vdk-plugins/vdk-trino/src/vdk/plugin/trino/trino_connection.py

```diff
@@ -109,61 +107,46 @@ def execute_query_with_retries(self, query):
         res = super().execute_query(query)
         return res
 
-    def add_num_rows_after_query(self, lineage_data, res):
-        if lineage_data:
-            lineage_data["output_num_rows_after"] = self._get_output_table_num_rows(
-                lineage_data
-            )
-            if "outputTable" in lineage_data:
-                if res and res[0] and res[0][0]:
-                    lineage_data["output_num_rows_updated"] = res[0][0]
-
     def _get_lineage_data(self, query, query_id):
-        if self._lineage_logger:
+
+        from vdk.plugin.trino import lineage_utils
+        import sqlparse
+
+        statement = sqlparse.parse(query)[0]
+
+        if statement.get_type() == "ALTER":
+            rename_table_names = lineage_utils.parse_rename_table_names(query)
+            if rename_table_names:
+                return {
+                    "@id": query_id,
+                    "@type": "rename_table",
+                    "table_from": rename_table_names[0],
+                    "table_to": rename_table_names[1],
+                    "query": query,
+                    "status": "OK",
+                }
+
+        elif statement.get_type() == "SELECT" or statement.get_type() == "INSERT":
             try:
-                with closing_noexcept_on_close(self.__cursor()) as cur:
+                with closing_noexcept_on_close(self._cursor()) as cur:
                     cur.execute(f"EXPLAIN (TYPE IO, FORMAT JSON) {query}")
                     result = cur.fetchall()
                     if result:
                         import json
 
                         data = json.loads(result[0][0])
-                        data["@type"] = "taurus_query_io"
+                        data["@type"] = "insert_select"
```

Comment:
Why not specify the concrete type returned by sqlparse - "select" "insert" , and above "alter" ?

Reply:
There's one more type  "insert into <table> select from <table>" which is a combination. So basically insert and select queries can be considered an insert_select statement with no input or no output. Just wanted to make things simpler. Do you think it makes sense to have strictly support for 4 operations:
- insert
- select
- insert_select
- rename_table (alter is too generic)

Reply:
We can have insert select (`INSERT INTO orders SELECT * FROM new_orders;`
or insert values (`INSERT INTO cities VALUES (1, 'San Francisco');`) 

You want to differentiate them ?

Reply:
And yes we can support (for now) those operations only